scripts/scale-test/setup_cluster.py

```diff
@@ -0,0 +1,237 @@
+import json
+import logging
+import subprocess
+from argparse import ArgumentParser, Namespace
+from typing import List
+
+from benchmarks import juju
+from benchmarks.models import Cluster, Unit
+from benchmarks.utils import timeit
+
+APPLICATION = "microk8s-node"
+DEFAULT_CHANNEL = "1.24/stable"
+
+logging.basicConfig(format="[%(levelname)s]: %(message)s", level=logging.INFO)
+
+
+@timeit
+def install_microk8s(model, units: List[Unit], channel=DEFAULT_CHANNEL):
+    configure_proxy(units)
+    reboot_and_wait(model)
+    install_snap(channel)
+    update_etc_hosts(units)
```

Comment:
We will also need to configure containerd to use specific docker credentials https://microk8s.io/docs/dockerhub-limits .